Podcast/DownloadManager.swift

```diff
@@ -24,80 +34,153 @@ class DownloadManager: NSObject {
     }
     
     var downloaded: [String: Episode]
+    var downloadRequests: [String: DownloadRequest]
+    var resumeData: [String: Data]
+    
+    // QUESTION: delegate or NSNotification or custom KVO?
+    // Hopefully only temporary, we'll see what realm does.
+    // This kind of an abuse of delegates but honestly it's
+    // kind of nice. Basically any view that wants updates
+    // sets itself as the delegate on viewDidAppear.
+    // Doesn't have to worry about removing itself because
+    // it's a weak reference, and only one view at a time
+    // gets updates.
+    weak var delegate: EpisodeDownloader?
```

Comment:
interesting... I think it's fine to have it as this now before Realm / it would require another rewrite of downloading again which prob isn't worth it if we're switching to Realm anyway

Reply:
Yeah Realm shouldn't require too much of a rewrite, but we need something until then.